### PR TITLE
Feature: extend barcode collection search

### PR DESCRIPTION
**Pull request by marcomoscasgr**
Metadata updates works on ONT paths uploaded from the instrument to iRODS. This PR will add:
- recursive search of barcode collections on which apply metadata for rebasecalled data
- new tests for old and new rebasecalled and plexed data
- two new test dataset: old and new plexed and rebasecalled folder structures
- test updates for mlwh queries and html meta-report file

### CODE REVIEW

File: tests/ont/conftest.py [hunk 1]
```diff
@@ -137,9 +138,9 @@ def make_mplex_flowcell(ex, sl, tid, bc, n):
             study=study_z,
             instrument_name=instrument_name,
             instrument_slot=sl,
-            experiment_name=f"multiplexed_experiment_{ex :0>3}",
+            experiment_name=f"{ex_name}_{ex_n :0>3}",
             id_lims=f"Example LIMS ID {n}",
-            id_flowcell_lims=f"flowcell{sl + 100 :0>3}",
+            id_flowcell_lims=f"flowcell{sl + fc_start :0>3}",
```
Comment (kjsanger) [inline tests/ont/conftest.py:143]: Black doesn't catch this, but PEP 8 says there should not be whitespace before the `:` on the format string

File: tests/ont/conftest.py [hunk 2]
```diff
@@ -245,8 +283,57 @@ def ont_synthetic_irods(tmp_path):
             ]
             coll.add_metadata(*meta)
 
-    # We have synthetic data only for simple_experiment_001 and
-    # multiplexed_experiment_001
+    for expt in range(1, 2):
+        for slot in range(1, 2):
+            expt_name = f"old_rebasecalled_multiplexed_experiment_{expt :0>3}"
+            id_flowcell = f"flowcell{slot + 200 :0>3}"
```
Comment (kjsanger) [inline tests/ont/conftest.py:289]: As above (and more below).

File: tests/ont/test_metadata_update.py [hunk 3]
```diff
@@ -555,3 +699,87 @@ def test_retains_consent_withdrawn_mx(
...
+        expected_bcolls = 3
+        root_path = PurePath(
+            "/testZone/home/irods/test/ont_synthetic_irods/synthetic/barcode_collection_test"
+        )
+        expt = "multiplexed_folder_experiment_004"
+        path = root_path / expt / "20190904_1514_GA10000_flowcell404_fg345hil"
+        expected_tag_identifiers = [
+            ont_tag_identifier(tag_index) for tag_index in range(1, 6)
+        ]
+        actual_tag_identifies = [
```
Comment (kjsanger) [inline tests/ont/test_metadata_update.py:776]: Typo

**Review by kjsanger — CHANGES_REQUESTED** [review]

File: src/npg_irods/ont.py [hunk 4]
```diff
@@ -537,10 +537,7 @@ def barcode_collections(coll: Collection, *tag_identifier) -> list[Collection]:
                 # so possibly this was not deplexed on-instrument for some reason e.g.
                 # a non-standard tag set was used
                 log.warn(
-                    "No barcode sub-collection",
-                    parent=parent,
-                    subfolder=barcode_name_from_id(tag_id),
-                    tag_identifier=tag_id,
+                    "No barcode sub-collection", path=str(bpath), tag_identifier=tag_id
```
Comment (kjsanger) [inline src/npg_irods/ont.py]: I'm pretty sure the `str()` call is unnecessary (also on other cases above) and it reads better without.

Comment (marcomoscasgr) [reply src/npg_irods/ont.py]: The last push includes this change.

Comment (marcomoscasgr) [reply src/npg_irods/ont.py]: I have just spotted the other in the string above that. Thanks

**Review by kjsanger — APPROVED** [review]
There's a lot of commits here, so I'll squash them to make the change clearer

**Pull request merged**